cpp/src/arrow/array/validate.cc

```diff
@@ -985,10 +985,23 @@ Status ValidateArrayFull(const Array& array) { return ValidateArrayFull(*array.d
 
 ARROW_EXPORT
 Status ValidateUTF8(const ArrayData& data) {
-  DCHECK(data.type->id() == Type::STRING || data.type->id() == Type::STRING_VIEW ||
-         data.type->id() == Type::LARGE_STRING);
-  UTF8DataValidator validator{data};
-  return VisitTypeInline(*data.type, &validator);
+  if (data.type->id() == Type::EXTENSION) {
```

Comment:
Did you add tests for this change?

Reply:
Added a `TEST_F(UTF8ExtensionArrayTest, JSONExtensionType)` test to `utf8_util_test.cc`.